src/toil/provisioners/clusterScaler.py

```diff
@@ -331,6 +356,19 @@ def tryRun(self):
                                 estimatedNodes)
                     totalNodes = self.scaler.provisioner.setNodeCount(numNodes=estimatedNodes,
                                                                       preemptable=self.preemptable)
+                    
+                    # If we were scaling up the number of preemptable nodes and failed to meet
+                    # our target, we need to update the slack so that non-preemptable nodes will
+                    # be allocated instead and we won't block. If we _did_ meet our target,
+                    # we need to reset the slack to 0.
+                    if self.preemptable:
+                        if totalNodes < estimatedNodes:
+                            deficit = estimatedNodes - totalNodes
+                            logger.debug('Preemptable scaler detected deficit of %d.', deficit)
```

Comment:
I would probably make this message more descriptive and promote it to `logger.info`. IMO, it is useful to the end user to know that their bid price was insufficient to get preemptable nodes (or, that there wasn't sufficient capacity). Ignore this if we're logging something similar elsewhere; we may be.